native/src/main/java/io/ballerina/stdlib/http/api/service/signature/AllQueryParams.java

```diff
@@ -90,13 +90,15 @@ public void populateFeed(HttpCarbonMessage httpCarbonMessage, ParamHandler param
                     parsedQueryValue = castParam(queryParam.getEffectiveTypeTag(),
                             queryValueArr.getBString(0).getValue());
                 }
-                paramFeed[index++] = ValueUtils.convert(parsedQueryValue, queryParam.getOriginalType());
-                paramFeed[index] = true;
+                castedQueryValue = ValueUtils.convert(parsedQueryValue, queryParam.getOriginalType());
             } catch (Exception ex) {
                 String message = "error in casting query param : '" + token + "'";
                 throw HttpUtil.createHttpStatusCodeError(QUERY_PARAM_BINDING_ERROR, message, null,
                         HttpUtil.createError(ex));
             }
+
+            paramFeed[index++] = queryParam.constraintValidation(castedQueryValue);
```

Comment:
```suggestion
            paramFeed[index++] = queryParam.validateConstraints(castedQueryValue);
```
matches better right?

Reply:
+1

Reply:
Fixed via [b2aa353](https://github.com/ballerina-platform/module-ballerina-http/pull/1608/commits/b2aa353bb917729ab479676d8b1a64081ccd94f0)